### PR TITLE
Choose fields labdiags & improve headers

### PR DESCRIPTION
**Pull request by MaxThevenet**
With this PR, the user can specify which fields to dump to back transformed diagnostics.
For this, it also moves Fortran functions `warpx_copy_slice_2/3d` to C+.
Finally, the Back-transformed diagnostic headers should be self-consistent now. Script `read_raw_data.py` was modified accordingly.

### CODE REVIEW

**Comment by MaxThevenet** [comment]
This PR is now ready for review.

File: Source/Diagnostics/BoostedFrameDiagnostic.cpp [hunk 1]
```diff
@@ -446,6 +444,36 @@ namespace
 }
 #endif
 
+namespace
+{
+    void
+    CopySlice(MultiFab& tmp, MultiFab& buf, int k_lab, 
+              std::vector<int> map_actual_fields_to_dump)
```
Comment (atmyers) [inline Source/Diagnostics/BoostedFrameDiagnostic.cpp]: It's probably better to pass this vector by const reference.

File: Source/Diagnostics/BoostedFrameDiagnostic.cpp [hunk 2]
```diff
@@ -446,6 +444,36 @@ namespace
...
+        const int ncomp_to_dump = map_actual_fields_to_dump.size();
+        // Copy data from MultiFab tmp to MultiFab data_buffer[i].
+        for (MFIter mfi(tmp, TilingIfNotGPU()); mfi.isValid(); ++mfi) {
+            Array4<      Real> tmp_arr = tmp[mfi].array();
+            Array4<      Real> buf_arr = buf[mfi].array();
+            // For 3D runs, tmp is a 2D (x,y) multifab, that contains only 
+            // slice to write to file.
+            const Box& bx  = mfi.tilebox();
+            
+            ParallelFor(bx, ncomp_to_dump,
```
Comment (atmyers) [inline Source/Diagnostics/BoostedFrameDiagnostic.cpp]: Did you verify that this works for both 2D and 3D?

File: Source/Diagnostics/BoostedFrameDiagnostic.cpp [hunk 3]
```diff
@@ -844,20 +901,26 @@ writeMetaData ()
         HeaderFile << dt_snapshots_lab_ << "\n";    
         HeaderFile << gamma_boost_ << "\n";
         HeaderFile << beta_boost_ << "\n";
-        HeaderFile << dz_lab_ << "\n";
```
Comment (atmyers) [inline Source/Diagnostics/BoostedFrameDiagnostic.cpp]: Could you also make sure that none of these changes break the USE_HDF5 option?

File: Source/Diagnostics/BoostedFrameDiagnostic.H [hunk 4]
```diff
@@ -32,17 +33,26 @@ class BoostedFrameDiagnostic {
         
         std::string file_name;
         amrex::Real t_lab;
-        amrex::Real zmin_lab;
-        amrex::Real zmax_lab;
+        // amrex::Real zmin_lab;
+        // amrex::Real zmax_lab;
+        amrex::RealBox prob_domain_;
+        amrex::IntVect prob_ncells_;
```
[the remark this answers is not in the document]
Comment (MaxThevenet) [reply Source/Diagnostics/BoostedFrameDiagnostic.H]: I did so in the last commit

File: Source/Diagnostics/BoostedFrameDiagnostic.cpp (hunk 1 above)
Comment (MaxThevenet) [reply Source/Diagnostics/BoostedFrameDiagnostic.cpp]: I changed `std::vector<int> map_actual_fields_to_dump` to `const std::vector<int>& map_actual_fields_to_dump` in the last commit. Is this what you meant?

File: Source/Diagnostics/BoostedFrameDiagnostic.cpp (hunk 2 above)
Comment (MaxThevenet) [reply Source/Diagnostics/BoostedFrameDiagnostic.cpp]: Yep, it does.

File: Source/Diagnostics/BoostedFrameDiagnostic.cpp (hunk 3 above)
Comment (MaxThevenet) [reply Source/Diagnostics/BoostedFrameDiagnostic.cpp]: Would you have a test that I could use (and even add to automated tests) for that?

**Comment by MaxThevenet** [comment]
@atmyers Thank you for your review, I answered your comments and made a commit. Concerning hdf5 tests, are there some instructions on how to compile WarpX with hdf5 support?

**Comment by atmyers** [comment]
You compile with `make -j4 USE_HDF5=TRUE` - you'll probably need to realclean first. Probably the easiest way to test is to use cori and load the hdf5-parallel module. I mainly just want to make sure that none of the removed variables are used by that capability.

**Comment by MaxThevenet** [comment]
I had left a typo for the HDF5 compilation indeed, thanks!
Now
```
import h5py
f = h5py.File('HDF5_lab_frame_data/snapshot00003', 'r')
list(f.keys())
```
returns
```
['Bx', 'Ey', 'beam', 'electrons', 'ions']
```
and the fields are correct.

**Pull request merged**